internal/private/private.go

```diff
@@ -117,14 +120,15 @@ type TryReusingBlobOptions struct {
 	EmptyLayer          bool                   // True if the blob is an "empty"/"throwaway" layer, and may not necessarily be physically represented.
 	LayerIndex          *int                   // If the blob is a layer, a zero-based index of the layer within the image; nil otherwise.
 	SrcRef              reference.Named        // A reference to the source image that contains the input blob.
-	TOCDigest           *digest.Digest         // If specified, the blob can be looked up in the destination also by its TOC digest.
+	TOCDigest           digest.Digest          // If specified, the blob can be looked up in the destination also by its TOC digest.
 }
 
 // ReusedBlob is information about a blob reused in a destination.
 // It is the subset of types.BlobInfo fields the transport is responsible for setting.
 type ReusedBlob struct {
-	Digest digest.Digest // Must be provided
-	Size   int64         // Must be provided
+	Digest    digest.Digest // Must be provided, can be empty if TOCDigest is present
+	TOCDigest digest.Digest // Must be provided, can be empty if Digest is present
```

Comment:
See elsewhere, I’m not sure this works.

In particular, in all the manifest formats, `digest` is a mandatory value. We have to write _something_ in there even if we find/match a TOC.

Also, below, `Size` of the full blob must be known; so there must conceptually be a full blob and a `Digest` value exists.

(Yes, the c/image/storage model of `PutBlob` pretending to record the original compressed blob is basically cheating, mostly but not always successfully.)

Reply:
Resolved in #2288 ; there is now a `MatchedByTOCDigest` sufficient for the existing user.